external-providers/java-external-provider/pkg/java_external_provider/dependency.go

```diff
@@ -109,11 +112,29 @@ func (p *javaServiceClient) GetDependencies(ctx context.Context) (map[uri.URI][]
 		return m, err
 	}
 
-	p.depsMutex.RLock()
-	val := p.depsCache
-	p.depsMutex.RUnlock()
-	if val != nil {
-		return val, nil
+	if pomFile := p.findPom(); pomFile != "" {
+		// Read pom and create a hash.
+		// if pom hash and depCache return cache
+		file, err := os.Open(pomFile)
+		if err != nil {
+			p.log.Error(err, "unable to open the pom file", "pom path", pomFile)
+			return nil, err
+		}
+		hash := sha256.New()
+		if _, err := io.Copy(hash, file); err != nil {
```

Comment:
Looks like `file` may not always be a file? From demo output: `0.errors.java-pomxml-dependencies = read /analyzer-lsp/examples/quarkus-1-6-2-jar-exploded: is a directory`

Reply:
Yeah, I think in the case of binaries, there is some special stuff, I don't need the behavior changed for that, so adding back the behavior for that and adding handling nested poms.